pennylane/templates/subroutines/adder.py

```diff
@@ -0,0 +1,119 @@
+
+    **Example**
+
+    Sum of two integers :math:`m=8` and :math:`k=5` modulo :math:`mod=15`. Note that to perform this sum using qml.Adder we need that :math:`m,k < mod`
+
+    .. code-block::
+    
+        m = 8
+        k = 5
+        mod = 15
+        wires_m =[0,1,2,3]
+        work_wires=[4,5]
+        dev = qml.device("default.qubit", shots=1)
+        @qml.qnode(dev)
+        def adder_modulo(m, k, mod, wires_m, work_wires):
+            # Function that performs m + k modulo mod in the computational basis
+            qml.BasisEmbedding(m, wires=wires_m)
+            qml.Adder(k, wires_m, mod, work_wires)
+            return qml.sample(wires=wires_m)
+
+    .. code-block:: pycon
+
+        >>> print(f"The ket representation of {m} + {k} mod {mod} is {adder_modulo(m, k, mod,wires_m, work_wires)}")
+        The ket representation of 8 + 5 mod 15 is [1 1 0 1]
+
+    We can see that the result [1 1 0 1] corresponds to 13, which comes from :math:`8+5=13 \longrightarrow 13 mod 15 = 13`.
+    """
+
+    grad_method = None
+
+    def __init__(
+        self, k, wires, mod=None, work_wires=None, id=None
+    ):  # pylint: disable=too-many-arguments
+        if mod == None:
+            mod = 2 ** (len(wires))
+        if work_wires != None:
+            if any(wire in work_wires for wire in wires):
+                raise ValueError("Any wire in work_wires should not be included in wires.")
+        else:
+            work_wires = [wires[-1] + 1, wires[-1] + 2]
+        self.hyperparameters["k"] = k
+        self.hyperparameters["mod"] = mod
+        self.hyperparameters["work_wires"] = qml.wires.Wires(work_wires)
+        super().__init__(wires=wires, id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    @staticmethod
+    def compute_decomposition(k, mod, work_wires, wires):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+        Args:
+            k (int): number that wants to be added
+            mod (int): modulo of the sum
+            work_wires (Sequence[int]): the auxiliary wires to use for the sum modulo :math:`mod` when :math:`mod \neq 2^{\textrm{len(wires)}}`
+            wires (Sequence[int]): the wires the operation acts on
+        Returns:
+            list[.Operator]: Decomposition of the operator
+
+        **Example**
+
+        >>> qml.Adder.compute_decomposition(k=2,mod=8,wires=[0,1,2],work_wires=[3,4])
+        [QFT(wires=[0, 1, 2]),
+        PhaseAdder(wires=[0, 1, 2]),
+        Adjoint(QFT(wires=[0, 1, 2]))]
+        """
+        op_list = []
+        if mod == 2 ** len(wires):
+            qft_wires = wires
+        else:
+            qft_wires = work_wires[:1] + wires
+        # we perform m+k modulo mod
```

Comment:
Please see [here](https://docs.pennylane.ai/en/stable/development/guide/documentation.html#commenting-code) for instructions on commenting. This comment should be removed.